dashboard/test/controllers/omniauth_callbacks_controller_test.rb

```diff
@@ -463,6 +463,27 @@ class OmniauthCallbacksControllerTest < ActionController::TestCase
     assert_equal user.uid, uid
   end
 
+  test 'login: google_oauth2 silently takes over unmigrated Google Classroom student with matching email' do
+    email = 'test@foo.xyz'
+    uid = '654321'
+    user = create(:student, email: email)
+    google_classroom_student = create(:student, :unmigrated_google_sso, uid: uid)
+    google_classroom_student.update!(hashed_email: '')
```

Comment:
Should we have a factory for this?

Reply:
yeah, good idea. i'll make a `google_classroom_imported_student` factory or something similar